In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("Stage 2 Environment Ready!")

Stage 2 Environment Ready!


In [3]:
import pandas as pd

In [4]:
jobs_df = pd.read_csv("../data/processed/jobs_master.csv")

In [5]:
jobs_df.head(10)

,job_id,job_title,industry,location,salary_usd,skills_required,remote_option,company_size,posting_date
0,1,Quantum Researcher,Quantum Computing,Singapore,175780,"linear algebra, quantum algorithms",No,Large,2025-07-22
1,2,Renewable Energy Engineer,Green Tech,Singapore,137481,"climate data analysis, energy modeling",Yes,Large,2025-09-26
2,3,Quantum Researcher,Quantum Computing,Tokyo,182081,"linear algebra, qiskit",No,Medium,2025-12-31
3,4,Sustainability Analyst,Green Tech,Singapore,113822,"climate data analysis, energy modeling",No,Large,2025-05-29
4,5,Smart Contract Engineer,Blockchain,London,92575,"rust, solidity",Yes,Small,2025-03-30
5,6,Smart Contract Engineer,Blockchain,Tokyo,173379,"solidity, rust",Yes,Medium,2025-08-10
6,7,Renewable Energy Engineer,Green Tech,New York,99659,"climate data analysis, energy modeling",No,Small,2025-05-27
7,8,Quantum Software Developer,Quantum Computing,London,210842,"qiskit, quantum algorithms",Yes,Large,2025-04-13
8,9,Sustainability Analyst,Green Tech,New York,189475,"climate data analysis, energy modeling",No,Small,2025-04-12
9,10,Blockchain Developer,Blockchain,Tokyo,228992,"ethereum, solidity",No,Large,2025-06-28


In [6]:
jobs_df.shape

(10000, 9)

In [7]:
jobs_df.columns.to_list()

['job_id',
 'job_title',
 'industry',
 'location',
 'salary_usd',
 'skills_required',
 'remote_option',
 'company_size',
 'posting_date']

In [8]:
jobs_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   job_id           10000 non-null  int64
 1   job_title        10000 non-null  str  
 2   industry         10000 non-null  str  
 3   location         10000 non-null  str  
 4   salary_usd       10000 non-null  int64
 5   skills_required  10000 non-null  str  
 6   remote_option    10000 non-null  str  
 7   company_size     10000 non-null  str  
 8   posting_date     10000 non-null  str  
dtypes: int64(2), str(7)
memory usage: 703.3 KB


In [9]:
jobs_df["posting_date"] = pd.to_datetime(
    jobs_df["posting_date"],
    errors="coerce"
)

In [10]:
jobs_df.info("posting_date")

<class 'pandas.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   job_id           10000 non-null  int64         
 1   job_title        10000 non-null  str           
 2   industry         10000 non-null  str           
 3   location         10000 non-null  str           
 4   salary_usd       10000 non-null  int64         
 5   skills_required  10000 non-null  str           
 6   remote_option    10000 non-null  str           
 7   company_size     10000 non-null  str           
 8   posting_date     10000 non-null  datetime64[us]
dtypes: datetime64[us](1), int64(2), str(6)
memory usage: 703.3 KB


In [11]:
jobs_df["posting_date"].dtype
jobs_df["posting_date"].isna().sum()

np.int64(0)

In [12]:
jobs_df.head()

,job_id,job_title,industry,location,salary_usd,skills_required,remote_option,company_size,posting_date
0,1,Quantum Researcher,Quantum Computing,Singapore,175780,"linear algebra, quantum algorithms",No,Large,2025-07-22
1,2,Renewable Energy Engineer,Green Tech,Singapore,137481,"climate data analysis, energy modeling",Yes,Large,2025-09-26
2,3,Quantum Researcher,Quantum Computing,Tokyo,182081,"linear algebra, qiskit",No,Medium,2025-12-31
3,4,Sustainability Analyst,Green Tech,Singapore,113822,"climate data analysis, energy modeling",No,Large,2025-05-29
4,5,Smart Contract Engineer,Blockchain,London,92575,"rust, solidity",Yes,Small,2025-03-30


In [13]:
jobs_df["posting_date"].dtype

dtype('<M8[us]')

# Dataset Overview

## Dataset Name
jobs_master.csv

## Number of rows
10000

## Number of columns
9

## Column Names       
- job_id           10000 non-null  int64         
- job_title        10000 non-null  str           
- industry         10000 non-null  str           
- location         10000 non-null  str           
- salary_usd       10000 non-null  int64         
- skills_required  10000 non-null  str           
- remote_option    10000 non-null  str           
- company_size     10000 non-null  str           
- posting_date     10000 non-null  datetime64[us]

## Initial observations
- posting_date changed to datetime from str.
- No duplicated available.
- No null values in the dataset.
- Ready for analysis.

In [14]:
jobs_df.isnull().sum()

job_id             0
job_title          0
industry           0
location           0
salary_usd         0
skills_required    0
remote_option      0
company_size       0
posting_date       0
dtype: int64

In [24]:
for column in jobs_df.columns:
    print(f"{column}: {jobs_df[column].nunique()} unique values")

job_id: 10000 unique values
job_title: 9 unique values
industry: 4 unique values
location: 6 unique values
salary_usd: 9738 unique values
skills_required: 20 unique values
remote_option: 2 unique values
company_size: 3 unique values
posting_date: 365 unique values


# Data Quality findings

## Missing values
- No missing values 
- No null columns
- No values were replaced

## Unique values
- job_id: 10000 unique values
- job_title: 9 unique values
- industry: 4 unique values
- location: 6 unique values
- salary_usd: 9738 unique values
- skills_required: 20 unique values
- remote_option: 2 unique values
- company_size: 3 unique values
- posting_date: 365 unique values